### PR TITLE
DAC trigger support

### PR DESCRIPTION
**Pull request by Adrian-Stanea**
Introduce  support for `START` and `STOP` trigger events in the M2KAnalogOut. 
Ensure predictable reset of the interface at 0V

**Features:**
- **Trigger events:**
     - **`START`:** initiates DAC output based on a trigger event. This trigger can be customized using the `rearm` functionality, allowing for different behaviors in response to each trigger event.
     - **`STOP:`** stops the DAC output, applicable only to cyclic buffers.

**Output Reset:**
Utilize the `raw` and `raw_enable` attributes to ensure outputs are reset to 0V. This is particularly important in scenarios where external programs may have altered the raw IIO attribute or when the DAC output remains in the last sample hold state.

**Demo Example:**
A Python demo has been added to showcase the START trigger capabilities

**NOTES:**
- **STOP Trigger:** the STOP trigger is functional only with cyclic buffers. In non-cyclic mode, if all samples are already sent and the channel is in the last sample hold state, the STOP trigger will have no impact.

- **START Trigger with Non-Cyclic Buffers:** When using the START trigger with non-cyclic buffers and with rearm functionality disabled, it is recommended to merge all buffers into a single buffer. This addresses potential unpredictable behavior due to race conditions in HDL. In this scenario, since all buffers will be sent regardless, merging them will standardize the outcome.

### CODE REVIEW

File: include/libm2k/analog/m2kanalogout.hpp [hunk 1]
```diff
@@ -486,6 +487,40 @@ class LIBM2K_API M2kAnalogOut
 	 * @return unsigned short - the value set in the raw attribute
 	 */
 	virtual unsigned short setVoltageRaw(unsigned int chn_idx, unsigned short raw) = 0;
+
+
+	/**
+	* @brief Get the hardware trigger handler
+	*
+	* @return A pointer to the hardware trigger trigger
```
Comment (AlexandraTrifan) [inline include/libm2k/analog/m2kanalogout.hpp:495]: duplicate 'trigger'

File: src/m2k_impl.cpp [hunk 2]
```diff
@@ -69,12 +70,20 @@ M2kImpl::M2kImpl(std::string uri, iio_context* ctx, std::string name, bool sync)
 	m_instancesPowerSupply.clear();
 
 	m_firmware_version = getFirmwareVersion();
-
-	int diff = Utils::compareVersions(m_firmware_version, "v0.24");
-	if (diff < 0) {	//m_firmware_version < 0.24
+	// Remove suffix from dev version
+    std::size_t pos = m_firmware_version.find("-dirty");
```
Comment (AlexandraTrifan) [inline src/m2k_impl.cpp:74]: fix indent

File: src/m2khardwaretrigger_v0.33_impl.cpp [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+#include <iio.h>
+
+using namespace libm2k;
+using namespace libm2k::utils;
+using namespace libm2k::digital;
+using namespace std;
+
+std::vector<std::string> M2kHardwareTriggerV033Impl::m_trigger_source_out = {
+	"none",
+	"trigger-Ti",
```
Comment (AlexandraTrifan) [inline src/m2khardwaretrigger_v0.33_impl.cpp:40]: is T intentionally uppercase?

**Comment by AlexandraTrifan** [comment]
otherwise looks good
i'll run some tests as well with the new fw

**Review by Adrian-Stanea — COMMENTED** [review]
> is T intentionally uppercase? 

The name was defined to match the attribute definition within the firmware.

**Review by AlexandraTrifan — APPROVED** [review]

**Pull request merged**